jujugui/static/gui/src/app/views/topology/service.js

```diff
@@ -1428,6 +1411,60 @@ YUI.add('juju-topology-service', function(Y) {
     createServiceNode: function(node, self) {
       node.attr({'data-name':  function(d) { return d.name; }});
 
+      // Draw a relation button.
+      var relationButton = node.filter(function(d) {
+        return d3.select(this)
+            .select('.relation-button').empty();
+      })
+          .insert('g', ':first-child')
+          .attr('class', 'relation-button')
+          .attr('transform', function(d) {
+            // Position the block so that the relation indicator will
+            // appear at the top.
+            return 'translate(' + [d.subordinate ? 65 : 95, 30] + ')';
+          });
+
+      relationButton.append('line')
+        .attr({
+          'x1': 0,
+          'y1': 0,
+          'x2': 0,
+          'y2': 30
+        })
+        .attr('stroke-width', 1)
+        .attr('stroke', '#888888');
+        relationButton.append('circle')
+        .attr({
+          'cx': 0,
+          'cy': 34,
+          'r': 4
+        })
+        .attr('fill', '#888888');
+
+      relationButton.append('circle')
+        .classed('relation-button__link', true)
+        .attr({
+          cx: 0,
+          cy: 0,
+          r: 15,
+          fill: '#f8f8f8',
+          stroke: '#888888',
+          'stroke-width': 1
```

Comment:
The circle is pretty janky, previously @anthonydillon was able to solve this by setting this to 1.1 instead of 1 which also does wonders here.